### PR TITLE
Doc and XSD Updates

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

File: docs/_writing_multiple_outputs.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            <data format="interval" name="output1" metadata_source="input1" />
+            <data format="pdf" name="output2" />
+        </outputs>
+    </tool>
+  
+---------------------------------------------------------
+Static Outputs Determinable from Inputs 
+---------------------------------------------------------
+
+In cases when the number of output files varies, but can be determined based upon a user's parameter selection, the use of the filter tag can be used.  The text contents of the ``<filter>`` tag are ``eval`` ed and if the expression is ``True`` a dataset will be created as normal.  If the expression is ``False``, the output dataset will not be created; instead a ``NoneDataset`` object will be created and made available. When used on the command line the text ``None`` will appear instead of a file path. The local namespace of the filter has been populated with the tool parameters.
```
Comment (mvdbeek) [inline docs/_writing_multiple_outputs.rst]: parameter selection, the use of the filter tag can be used. -> arameter selection, the filter tag can be used.

File: docs/_writing_multiple_outputs.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    </outputs>
+
+
+---------------------------------------------------------
+Single HTML Output
+---------------------------------------------------------
+
+There are times when a single history item is desired, but this history item is composed of multiple files which are only useful when considered together. This is done by having a single (``primary``) output and storing additional files in a directory (single-level) associated with the primary dataset.
+
+A common usage of this strategy is to have the primary dataset be an HTML file and then store additional content (reports, pdfs, images, etc) in the dataset extra files directory. The content of this directory can be referenced using relative links with in the primary (HTML) file, clicking on the eye icon to view the dataset will display the HTML page.
```
Comment (mvdbeek) [inline docs/_writing_multiple_outputs.rst]: s/with in/within/

File: docs/_writing_multiple_outputs.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+The application must create and write valid html to setup the page ``$html_file`` seen by the user when they view (eye icon) the file. It must create and write that new file at the path passed by Galaxy as the ``$html_file`` command line parameter. All application outputs that will be included as links in that html code should be placed in the specific output directory ``$html_file.extra_files_path`` passed on the command line. The external application is responsible for creating that directory before writing images and files into it. When generating the html, The files written by the application to ``$html_file.extra_files_path`` are referenced in links directly by their name, without any other path decoration - eg:
+
+
+.. code-block:: xml
+
+    <a href="file1.xls">Some special output</a>
+    <br/>
+    <img src="image1.jpg" >
+
+The (now unmaintained) Galaxy Tool Factory includes code to gather all output files and create a page with links and clickable PDF thumbnail images which may be useful as a starting point - eg see https://bitbucket.org/fubar/rgalaxy/src/9932187787e592238c2c6fb514a39ff3a705a9af/tools/rgenetics/rgToolFactory.py?at=default
```
Comment (mvdbeek) [inline docs/_writing_multiple_outputs.rst]: Perhaps link to here: https://github.com/galaxyproject/tools-iuc/blob/master/tools/tool_factory_2/rgToolFactory2.py#L741 ?

File: docs/_writing_multiple_outputs.rst [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+``galaxy.datatypes.text.Html`` (the ``html`` datatype) is a subclass of composite datasets so new subclasses of composite can be used to implement even more specific structured outputs but this requires adding the new definition to Galaxy - whereas Html files require no extension of the core framework. For more information visit `Composite Datatypes <https://wiki.galaxyproject.org/Admin/Datatypes/Composite%20Datatypes>`__. 
+
+
+---------------------------------------------------------
+Dynamic Numbers of Outputs
+---------------------------------------------------------
+
+This section discusses the case where the number of output datasets cannot be determined until the tool run is complete. If the outputs can be broken into
+groups or collections of similar/homogenous datasets - this is possibly a
+case for using dataset collections. If insted the outputs should be treated
```
Comment (mvdbeek) [inline docs/_writing_multiple_outputs.rst]: s/insted/instead/

File: docs/_writing_multiple_outputs.rst [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+Individual Datasets
+---------------------------------------------------------
+
+There are times when the number of output datasets varies entirely based upon the content of an input dataset and the user needs to see all of these outputs as new individual history items rather than as a collection of datasets or a group of related files linked in a single new HTML page in the history. Tools can optionally describe how to "discover" an arbitrary number of files that will be added after the job's completion to the user's history as new datasets. Whenever possible, one of the above strategies should be used instead since these discovered datasets cannot be used with workflows and require the user to refresh their history before they are shown.
+
+Discovering datasets (arbitrarily) require a fixed "parent" output dataset to key on - this dataset will act as the reference for our additional datasets. Sometimes the parent dataset that should be used makes sense from context but in instances where one does not readily make sense tool authors can just create an arbitrary text output (like a report of the dataset generation).
```
Comment (mvdbeek) [inline docs/_writing_multiple_outputs.rst]: s/require/requires/

**Comment by mvdbeek** [comment]
Excellent job, this is a good resource @jmchilton.

**Comment by jmchilton** [comment]
Thanks for the review @mvdbeek - I've incorporated your comments into the pull request.

I'm just waiting on one passing Travis check for any of these.

**Pull request merged**